pinot-query-planner/src/main/java/org/apache/calcite/sql/fun/PinotOperatorTable.java

```diff
@@ -87,13 +86,6 @@ public final void initNoDuplicate() {
     }
   }
 
-  private boolean notRegistered(SqlFunction op) {
-    List<SqlOperator> operatorList = new ArrayList<>();
-    lookupOperatorOverloads(op.getNameAsId(), op.getFunctionType(), op.getSyntax(), operatorList,
-        SqlNameMatchers.withCaseSensitive(false));
-    return operatorList.size() == 0;
-  }
```

Comment:
no please do not remove this registration check. passing `op.getFunctionType()` is not the same as passing directly `null`

it is being ignored by calcite right now but it is not guarantee not to be used in the future. deviating from calcite behavior could cause debuggability issues in the future (as we are basically overriding a behavior on our own)